src/launch/launcher.c

```diff
@@ -1434,6 +1435,22 @@ int launcher_run(Launcher *launcher) {
                         return error_fold(r);
         }
 
+        /*
+         * When a Condition*= setting fails, the JobRemoved signal normally returns
+         * 'done' which we interpret as a success, but it means we will wait in vain
+         * for the service to start. It cannot be changed without breaking backward
+         * compatibility, so a new StartUnitWithFlags was added to change it on-demand.
+         * Let's use it if it is available, and fallback to the old StartUnit otherwise.
+         */
+        r = object_supports_interface(launcher,
+                                      "org.freedesktop.systemd1",
+                                      "/org/freedesktop/systemd1",
+                                      "StartUnitWithFlags");
+        if (r < 0)
+                error_origin(r); /* Log but continue with backward-compatible method */
```

Comment:
This should be `return error_fold(r)`. If we wanted to make it a soft error, we should define and return a positive error code (see `util/error.c`), but as mentioned above, I think we should make this a hard error, so that point is moot.

Reply:
This means we'll also fail to run in case there's a problem with connecting to the bus and calling Introspectable, not just with malformed XML. Is that safe to do? I don't mind either way, was just trying to play it on the safe side, given it's backward compatible.

Reply:
updated as requested for now, if you change your mind let me know and I'll change it back

Reply:
I figure if we cannot talk to pid1 we'll have bigger problems (and we try to err on the side of failing early as a general rule). Though let's see if @dvdhrm has any other opinions.

Reply:
this also affects user sessions and buses, not just pid1/system bus (if it makes any difference)